packages/offix-datastore/src/replication/ReplicationEngine.ts

```diff
@@ -0,0 +1,42 @@
+import { OffixScheduler } from "offix-scheduler";
+
+import { IReplicator } from "./Replicator";
+import { Storage, StoreChangeEvent } from "../storage";
+
+/**
+ * Schedules replication events and handles replication errors
+ */
+export class ReplicationEngine {
+    private api: IReplicator;
+    private scheduler: Promise<OffixScheduler<StoreChangeEvent>>;
+    private storage: Storage;
+
+    constructor(
+        api: IReplicator,
+        storage: Storage
+    ) {
+        this.api = api;
+        this.storage = storage;
+        this.scheduler = new Promise((resolve, reject) => {
+            const scheduler = new OffixScheduler<StoreChangeEvent>({
```

Comment:
This looks really ugly. We need scheduler to change. There is no need for init for scheduler in new API etc.

Reply:
Again this is more on core team shoulders to improve scheduler. Gianluca works on this and we can see API changed.

Reply:
Yeah agreed, this is something that we've needed to improve for a while now.